tests/41end-to-end-keys/08-cross-signing.pl

```diff
@@ -682,11 +682,21 @@
       })->then( sub {
          sync_until_user_in_device_list( $user1, $user2 );
       })->then( sub {
-         matrix_get_e2e_keys( $user1, $user2_id );
+         retry_until_success {
+            # Wait until user1 sees signatures uploaded by user2.
+            # We retry_until_success because this is sensitive to replication delays.
+            matrix_get_e2e_keys( $user1, $user2_id )->then( sub {
+               my ( $content ) = @_;
+               log_if_fail "key query content2", $content;
+               $content->{device_keys}->{$user2_id}->{$user2_device}->{"signatures"}
```

Comment:
what you wrote here is entirely valid, but the `->` is implied between two subscripts (https://perldoc.perl.org/perlreftut#Arrow-Rule), and convention is to drop them.

```suggestion
               $content->{device_keys}{$user2_id}{$user2_device}{"signatures"}
```